src/OpenDB/test/gcd_abstract_lef.lefok

```diff
@@ -0,0 +1,812 @@
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  71.56 0 71.7 0.14 ;
+    END
+  END resp_msg[0]
+  PIN resp_msg[10]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal3 ;
+        RECT  200.12 73.99 200.26 74.13 ;
+    END
+  END resp_msg[10]
+  PIN resp_msg[11]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal3 ;
+        RECT  200.12 37.59 200.26 37.73 ;
+    END
+  END resp_msg[11]
+  PIN resp_msg[12]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  76.88 0 77.02 0.14 ;
+    END
+  END resp_msg[12]
+  PIN resp_msg[13]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  89.04 0 89.18 0.14 ;
+    END
+  END resp_msg[13]
+  PIN resp_msg[14]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  63.2 0 63.34 0.14 ;
+    END
+  END resp_msg[14]
+  PIN resp_msg[15]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  67.76 0 67.9 0.14 ;
+    END
+  END resp_msg[15]
+  PIN resp_msg[1]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal3 ;
+        RECT  0 114.87 0.14 115.01 ;
+    END
+  END resp_msg[1]
+  PIN resp_msg[2]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  77.64 0 77.78 0.14 ;
+    END
+  END resp_msg[2]
+  PIN resp_msg[3]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  92.84 201.46 92.98 201.6 ;
+    END
+  END resp_msg[3]
+  PIN resp_msg[4]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  117.16 201.46 117.3 201.6 ;
+    END
+  END resp_msg[4]
+  PIN resp_msg[5]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  130.08 201.46 130.22 201.6 ;
+    END
+  END resp_msg[5]
+  PIN resp_msg[6]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  117.92 201.46 118.06 201.6 ;
+    END
+  END resp_msg[6]
+  PIN resp_msg[7]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal3 ;
+        RECT  0 135.03 0.14 135.17 ;
+    END
+  END resp_msg[7]
+  PIN resp_msg[8]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  103.48 0 103.62 0.14 ;
+    END
+  END resp_msg[8]
+  PIN resp_msg[9]
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal2 ;
+        RECT  105.76 0 105.9 0.14 ;
+    END
+  END resp_msg[9]
+  PIN resp_rdy
+    DIRECTION INPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal3 ;
+        RECT  0 102.55 0.14 102.69 ;
+    END
+  END resp_rdy
+  PIN resp_val
+    DIRECTION OUTPUT ;
+    USE SIGNAL ;
+    PORT
+      LAYER metal3 ;
+        RECT  0 101.43 0.14 101.57 ;
+    END
+  END resp_val
+  OBS
+    LAYER metal1 ;
+    RECT  0 0 200.26 201.6 ;
+    LAYER metal4 ;
+    RECT  0 0 200.26 201.6 ;
+    LAYER metal7 ;
+    RECT  0 0 200.26 201.6 ;
+  END
```

Comment:
Wait, the obstruction is only on layers 1,4, and 7? That can't be right...

Reply:
I no longer block layers if there are only power nets on that layer.

Reply:
Looking at the input def there is metal on atleast metal1..6 from routing, how come those layers are not OBS? So that looks to be wrong.

Reply:
> if the power nets don't extend to the end of the block and they reside on the top most layer they wouldn't be accessible.

It is okay to assume this condition should never happen. If it does then it's a designer error. Macros should either not occupy the top layer (so that pins can be dropped from above) or they should be exposed on an edge for connection.

> I figured as long as long as there's only power nets on a layer it should be fine to allow pin access to them.

True, but it looks like it is dropping obstructions for normal routing.